ksql-engine/src/main/java/io/confluent/ksql/util/ExpressionTypeManager.java

```diff
@@ -74,27 +77,31 @@ public void setSchema(Schema schema) {
   @Override
   protected Expression visitArithmeticBinary(final ArithmeticBinaryExpression node,
                                              final ExpressionTypeContext expressionTypeContext) {
+    final int argCount = functionArguments.numCurrentFunctionArguments();
     process(node.getLeft(), expressionTypeContext);
     Schema leftType = expressionTypeContext.getSchema();
     process(node.getRight(), expressionTypeContext);
     Schema rightType = expressionTypeContext.getSchema();
     expressionTypeContext.setSchema(resolveArithmaticType(leftType, rightType));
+    if (functionArguments.numCurrentFunctionArguments() > argCount + 1) {
```

Comment:
When would this be false? Shouldn't each call to process add one argument to functionArguments?

Reply:
It might not have started visiting a function